src/tooltip.css.ts

```diff
@@ -1,177 +1,117 @@
 const enterTransform = createVar();
-const enterDoneTransform = createVar();
-const exitTransform = createVar();
-const enterActiveAnimationName = createVar();
-const arrowBoxShadow = createVar();
+const maxWidth = createVar();
 
-export const vars = {
-    shadowAlpha,
+export const tooltipVars = {
+    top,
+    left,
+    padding,
+    delay,
     enterTransform,
-    enterDoneTransform,
-    exitTransform,
-    enterActiveAnimationName,
-    arrowBoxShadow,
+    maxWidth,
 };
 
-export const fadeInBottomKeyframes = keyframes({
-    from: {opacity: 0},
-    '40%': {opacity: 1},
-    to: {
+export const tooltipTransitionClasses = {
+    entering: {
         opacity: 1,
-        transform: 'translateY(0)',
     },
-});
-
-export const fadeInTopKeyframes = keyframes({
-    from: {opacity: 0},
-    '40%': {opacity: 1},
-    to: {
+    entered: {
         opacity: 1,
-        transform: 'translateY(-100%)',
     },
-});
-
-export const fadeInXKeyframes = keyframes({
-    from: {opacity: 0},
-    '40%': {opacity: 1},
-    to: {
-        opacity: 1,
-        transform: 'translateX(0) translateY(-50%)',
+    exiting: {
+        opacity: 0,
     },
-});
-
-export const fadeOutKeyframes = keyframes({
-    from: {opacity: 1},
-    to: {opacity: 0},
-});
+    exited: {
+        opacity: 0,
+    },
+    unmounted: {},
+};
 
-export const arrow = style([
+export const container = style([
     sprinkles({
         position: 'absolute',
-        top: 0,
-        width: arrowSize,
-        height: arrowSize,
-        borderRadius: 2,
-        background: skinVars.colors.backgroundContainer,
-        border: 'regular',
     }),
     {
-        left: '50%',
-        transform: 'translate(-50%, -50%) rotate(45deg)',
-        boxShadow: arrowBoxShadow,
+        top,
+        left,
+        padding,
+        zIndex: 1,
+        opacity: 1,
+        transform: enterTransform,
+        filter: 'drop-shadow(0px 2px 2px rgba(0,0,0,0.2))',
     },
 ]);
 
-export const arrowTop = style({
-    top: '100%',
-    left: '50%',
-    transform: 'translateX(-50%)',
-});
-
-export const arrowRight = style({
-    right: '100%',
-    top: '50%',
-    transform: 'translateY(-100%) rotate(90deg)',
-    transformOrigin: 'bottom',
+export const enterTransition = style({
+    transition: `opacity .1s linear ${delay},transform .3s cubic-bezier(0.215,0.61,0.335,1) ${delay}`,
 });
 
-export const arrowBottom = style({
-    bottom: '100%',
-    left: '50%',
-    transform: 'translateX(-50%) rotate(180deg)',
+export const exitTransition = style({
+    transition: `opacity .1s linear`,
 });
 
-export const arrowLeft = style({
-    left: '100%',
-    top: '50%',
-    transform: 'translateY(-100%) rotate(-90deg)',
-    transformOrigin: 'bottom',
-});
+export const tooltip = style([
+    sprinkles({
+        position: 'relative',
+        borderRadius: vars.borderRadii.popup,
+        background: vars.colors.backgroundContainer,
+    }),
+    {
+        boxSizing: 'border-box',
```

Comment:
add a comment explaining this is needed because you're setting width with inline styles

Reply:
added